### PR TITLE
bloggy1: added model for post blog and database connection

### PR DESCRIPTION
**Pull request by wojurkowski**

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -7,7 +7,14 @@
...
       </profile>
     </annotationProcessing>
   </component>
+  <component name="JavacSettings">
+    <option name="ADDITIONAL_OPTIONS_OVERRIDE">
+      <module name="Blogg" options="-parameters" />
+      <module name="Bloggy" options="-parameters" />
+    </option>
+  </component>
 </project>
```
Comment (TomaszLaski) [inline .idea/compiler.xml]: remove this file (add this please to .gitignore)

File: .idea/misc.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
       </list>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_19" default="true" project-jdk-name="19" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="17" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>
```
Comment (TomaszLaski) [inline .idea/misc.xml]: same here

File: .idea/uiDesigner.xml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      </item>
+      <item class="javax.swing.JToolBar$Separator" icon="/com/intellij/uiDesigner/icons/toolbarSeparator.svg" removable="false" auto-create-binding="false" can-attach-label="false">
+        <default-constraints vsize-policy="0" hsize-policy="0" anchor="0" fill="1" />
+      </item>
+      <item class="javax.swing.JScrollBar" icon="/com/intellij/uiDesigner/icons/scrollbar.svg" removable="false" auto-create-binding="true" can-attach-label="false">
+        <default-constraints vsize-policy="6" hsize-policy="0" anchor="0" fill="2" />
+      </item>
+    </group>
+  </component>
+</project>
```
Comment (TomaszLaski) [inline .idea/uiDesigner.xml]: and here

File: pom.xml [hunk 4]
```diff
@@ -1,27 +1,37 @@
...
 		<version>3.0.2</version>
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
-	<groupId>com.example</groupId>
-	<artifactId>Bloggy</artifactId>
+	<groupId>com.example.</groupId>
+	<artifactId>BlogTest</artifactId>
 	<version>0.0.1-SNAPSHOT</version>
-	<name>Bloggy</name>
+	<name>BlogTest</name>
```
Comment (TomaszLaski) [inline pom.xml]: Please keep proper project name

File: src/main/java/com/example/Bloggy/BloggyApplication.java [hunk 5]
```diff
@@ -2,20 +2,20 @@
...
 
   public static void main(String[] args) {
     SpringApplication.run(BloggyApplication.class, args);
   }
 
-  @GetMapping("/hello")
-  public String hello(@RequestParam(value = "name", defaultValue = "World") String name) {
-    return String.format("Hello %s!", name);
-  }
 }
```
Comment (TomaszLaski) [inline src/main/java/com/example/Bloggy/BloggyApplication.java:15]: please format properly code

File: src/main/java/com/example/Bloggy/controller/BlogPostController.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        post.setTags(tags);
+        post.setTitle(title);
+        post.setContent(content);
+
+        blogPostRepository.save(post);
+        return "Saves";
+    }
+    @GetMapping(path="/all")
+    public @ResponseBody Iterable<BlogPost> getAllPosts() {
+        // This returns a JSON or XML with the users
```
Comment (TomaszLaski) [inline src/main/java/com/example/Bloggy/controller/BlogPostController.java]: remove not needed comments, leave only complicated buissness logic comments or anything that might need explanation

File: src/main/java/com/example/Bloggy/model/BlogPost.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    private String title;
+    private String content;
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
```
Comment (TomaszLaski) [inline src/main/java/com/example/Bloggy/model/BlogPost.java]: do we need to set ID? or should it be set up by database?

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1 +1,6 @@
-
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.url=jdbc:mysql://${MYSQL_HOST:localhost}:3306/Bloggy
+spring.datasource.username=Kuba
+spring.datasource.password=1234
```
Comment (TomaszLaski) [inline src/main/resources/application.properties]: please remove login credentials from repo for your own sake

**Review by TomaszLaski — CHANGES_REQUESTED** [review]

File: .idea/compiler.xml (hunk 1 above)
Comment (BartlomiejMilewski) [reply .idea/compiler.xml]: Done, catalogue removed.
`git rm --cached idea/\*`
https://superuser.com/questions/1397199/how-to-ignore-a-tracked-file-in-git-without-deleting-it

File: .idea/misc.xml (hunk 2 above)
Comment (BartlomiejMilewski) [reply .idea/misc.xml]: Done.

File: .idea/uiDesigner.xml (hunk 3 above)
Comment (BartlomiejMilewski) [reply .idea/uiDesigner.xml]: Done.

File: pom.xml (hunk 4 above)
Comment (BartlomiejMilewski) [reply pom.xml]: Done, name changed.

File: src/main/java/com/example/Bloggy/BloggyApplication.java (hunk 5 above)
Comment (BartlomiejMilewski) [reply src/main/java/com/example/Bloggy/BloggyApplication.java:15]: Done, formatting adjusted.
`Ctrl+Alt+L` (Intellij)

File: src/main/java/com/example/Bloggy/controller/BlogPostController.java (hunk 6 above)
Comment (BartlomiejMilewski) [reply src/main/java/com/example/Bloggy/controller/BlogPostController.java]: Done, unnecessary comment removed.

File: src/main/java/com/example/Bloggy/model/BlogPost.java (hunk 7 above)
Comment (BartlomiejMilewski) [reply src/main/java/com/example/Bloggy/model/BlogPost.java]: SQL sets ID as intended. Setter removed.

File: src/main/resources/application.properties (hunk 8 above)
Comment (BartlomiejMilewski) [reply src/main/resources/application.properties]: Credentials removed. File added to .gitignore.
application.properties.example added to Git.

File: .idea/compiler.xml (hunk 1 above)
Comment (TomaszLaski) [reply .idea/compiler.xml]: good job!

File: src/main/resources/application.properties (hunk 8 above)
Comment (TomaszLaski) [reply src/main/resources/application.properties]: good job!

**Review by TomaszLaski — APPROVED** [review]

**Pull request merged**